DNN Platform/Library/Entities/Modules/Settings/StringBasedSettings.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using DotNetNuke.Common;
+
+namespace DotNetNuke.Entities.Modules.Settings
+{
+    public class StringBasedSettings : ISettingsStore
+    {
+        Dictionary<string, string> _cache;
+        Dictionary<string, Action> _dirty;
+        Func<string, string> _get;
+        Action<string, string> _set;
+
+        /// <param name="get">Function to get a value out of your setting store</param>
+        /// <param name="set">Action to save a value back to the setting store</param>
+        public StringBasedSettings(Func<string, string> get, Action<string, string> set)
+        {
+            Requires.NotNull("Get", get);
+            Requires.NotNull("Set", set);
+
+            _get = get;
+            _set = set;
+            _cache = new Dictionary<string, string>();
+            _dirty = new Dictionary<string, Action>();
+        }
+
+        // All changes to the settings are recorded and get only executed on demand 
+        public void Save()
+        {
+            foreach (var save in _dirty.Values) save();
+            _dirty.Clear();
```

Comment:
@SCullman I may not have been thinking correctly about how this is used.  If we don't expect it to be used in a `static` context, then that shouldn't matter.  We probably don't _want_ to encourage it to be used that way, so I'm fine with it staying as is.  Sorry for my confusion.